### DISCUSSION TITLE
Save and Exit maxes out Swap and Memory

### DISCUSSION THREAD

**Discussion by Skwangles** [stella-cv/stella_vslam#591] [Q&A] [answered]
I'm using Stella vSLAM on the latest Ubuntu Linux for a Google Street View indoors mapping project where mapping sessions are up to 20 minutes of footage (around 121,000 landmarks and 3,829 key frames).
This is building on a single map file with multiple loop closures

Whenever I finish and go to save, my swap (2.1 GB) and memory (16GB) gets maxed and causes my computer to freeze and subsequently kill the process.

Any ideas how I can prevent thrashing, and the high memory usage during saving?

Thanks

Edit: The mapping and tracking works flawlessly, it is purely when I come to save to a map output file

Edit: Message pack file sizes are 200MB+

**Comment by ymd-stella** [comment, marked as answer]
Try changing map_format to sqlite3.

https://github.com/stella-cv/stella_vslam/blob/0a1f13e2b2ee3f63a8dc40601e6e5d59a4663111/example/euroc/EuRoC_mono.yaml#L57-L58
